### PR TITLE
Fix bugs for Queen class.

### PR DESCRIPTION
**Pull request by TanTheMan768**
* Separate line 41 into two lines
* Change if statements to check for `null` instead of `undefined`
* Stop adding moves when encountering enemy piece

### CODE REVIEW

File: js/queen.js [hunk 1]
```diff
@@ -38,12 +38,17 @@ export class Queen extends Piece{
...
         let maxCols = board[0].length;
-        let minRows, minCols = 0;
+        let minRows = 0;
+        let minCols = 0;
         /**
          * Get row increasing moves
          */
         for (let i = this.rowPos + 1; i < maxRows; i++) {
-            if (board[i][this.colPos] !== undefined && board[i][this.colPos].getPlayerId() == this.playerId) {
+            if (board[i][this.colPos] && board[i][this.colPos].getPlayerId() == this.playerId) {
```
Comment (jbeBan) [inline js/queen.js]: `===` should be used instead of `==` on line `47`.

File: js/queen.js [hunk 2]
```diff
@@ -52,7 +57,11 @@ export class Queen extends Piece{
          * Get row decreasing moves
          */
         for (let i = this.rowPos - 1; i >= minRows; i--) {
-            if (board[i][this.colPos] !== undefined && board[i][this.colPos].getPlayerId() == this.playerId) {
+            if (board[i][this.colPos] && board[i][this.colPos].getPlayerId() == this.playerId) {
```
Comment (jbeBan) [inline js/queen.js]: `===` should be used instead of `==` on line `60`.

File: js/queen.js [hunk 3]
```diff
@@ -61,7 +70,11 @@ export class Queen extends Piece{
          * Get col increasing moves
          */
         for (let i = this.colPos + 1; i < maxCols; i++) {
-            if (board[this.rowPos][i] !== undefined && board[this.rowPos][i].getPlayerId() == this.playerId) {
+            if (board[this.rowPos][i] && board[this.rowPos][i].getPlayerId() == this.playerId) {
```
Comment (jbeBan) [inline js/queen.js]: `===` should be used instead of `==` on line `73`.

File: js/queen.js [hunk 4]
```diff
@@ -70,7 +83,11 @@ export class Queen extends Piece{
          * Get col decreasing moves
          */
         for (let i = this.colPos - 1; i >= minCols; i--) {
-            if (board[this.rowPos][i] !== undefined && board[this.rowPos][i].getPlayerId() == this.playerId) {
+            if (board[this.rowPos][i] && board[this.rowPos][i].getPlayerId() == this.playerId) {
```
Comment (jbeBan) [inline js/queen.js]: `===` should be used instead of `==` on line `86`.

File: js/queen.js [hunk 5]
```diff
@@ -79,7 +96,11 @@ export class Queen extends Piece{
          * Get row and col increasing moves
          */
         for (let i = this.rowPos + 1, y = this.colPos + 1; i < maxRows && y < maxCols; i++, y++) {
-            if (board[i][y] !== undefined && board[i][y].getPlayerId() == this.playerId) {
+            if (board[i][y] && board[i][y].getPlayerId() == this.playerId) {
```
Comment (jbeBan) [inline js/queen.js]: `===` should be used instead of `==` on line `99`.

File: js/queen.js [hunk 6]
```diff
@@ -89,7 +110,11 @@ export class Queen extends Piece{
          * Get row and col decreasing moves
          */
         for (let i = this.rowPos - 1, y = this.colPos - 1; i >= minRows && y >= minCols; i--, y--) {
-            if (board[i][y] !== undefined && board[i][y].getPlayerId() == this.playerId) {
+            if (board[i][y] && board[i][y].getPlayerId() == this.playerId) {
```
Comment (jbeBan) [inline js/queen.js]: `===` should be used instead of `==` on line `113`.

File: js/queen.js [hunk 7]
```diff
@@ -98,7 +123,11 @@ export class Queen extends Piece{
          * Get row increasing and col decreasing moves
          */
         for (let i = this.rowPos + 1, y = this.colPos - 1; i < maxRows && y >= minCols; i++, y--) {
-            if (board[i][y] !== undefined && board[i][y].getPlayerId() == this.playerId) {
+            if (board[i][y] && board[i][y].getPlayerId() == this.playerId) {
```
Comment (jbeBan) [inline js/queen.js]: `===` should be used instead of `==` on line `126`.

File: js/queen.js [hunk 8]
```diff
@@ -107,7 +136,11 @@ export class Queen extends Piece{
          * Get row decreasing and col increasing moves
          */
         for (let i = this.rowPos - 1, y = this.colPos + 1; i >= minRows && y < maxCols; i--, y++) {
-            if (board[i][y] !== undefined && board[i][y].getPlayerId() == this.playerId) {
+            if (board[i][y] && board[i][y].getPlayerId() == this.playerId) {
```
Comment (jbeBan) [inline js/queen.js]: `===` should be used instead of `==` on line `139`.

**Review by jbeBan — CHANGES_REQUESTED** [review]

File: js/queen.js (hunk 8 above)
Comment (TanTheMan768) [reply js/queen.js]: Updated.

File: js/queen.js (hunk 7 above)
Comment (TanTheMan768) [reply js/queen.js]: Updated.

File: js/queen.js (hunk 6 above)
Comment (TanTheMan768) [reply js/queen.js]: Updated.

File: js/queen.js (hunk 5 above)
Comment (TanTheMan768) [reply js/queen.js]: Updated.

File: js/queen.js (hunk 4 above)
Comment (TanTheMan768) [reply js/queen.js]: Updated.

File: js/queen.js (hunk 3 above)
Comment (TanTheMan768) [reply js/queen.js]: Updated.

File: js/queen.js (hunk 2 above)
Comment (TanTheMan768) [reply js/queen.js]: Updated.

File: js/queen.js (hunk 1 above)
Comment (TanTheMan768) [reply js/queen.js]: Updated.

**Review by jbeBan — APPROVED** [review]
Excellent!

**Pull request merged**